Src/Particle/AMReX_ParticleIO.H

```diff
@@ -1617,29 +1617,29 @@ ParticleContainer<NStructReal,NStructInt,NArrayReal, NArrayInt>::WriteCoarsenedA
                         {
                             
                             // Only keep particles in even cells               
-                            if (it->m_idata.id > 0) {
+                            if (it->id() > 0) {
                               
-                                File << it->m_idata.id  << ' ';
-                                File << it->m_idata.cpu << ' ';
+                                File << it->id()  << ' ';
+                                File << it->cpu() << ' ';
                               
-                                AMREX_D_TERM(File << it->m_rdata.pos[0] << ' ',
-                                                  << it->m_rdata.pos[1] << ' ',
-                                                  << it->m_rdata.pos[2] << ' ');
+                                AMREX_D_TERM(File << it->pos(0) << ' ',
+                                                  << it->pos(1) << ' ',
+                                                  << it->pos(2) << ' ');
                               
                                 for (int i = 0; i < NumRealComps(); i++) {
                                     File << soa.GetRealData(i)[index] << ' ';
                                 }
                                 index++;
                                 
-                                for (int i = AMREX_SPACEDIM; i < AMREX_SPACEDIM + NStructReal; i++) {
-                                    char ws = (i == AMREX_SPACEDIM + NStructReal - 1) ? '\n' : ' ';
-                                    if (i == AMREX_SPACEDIM) {
+                                for (int i = 0; i < NStructReal; i++) {
+                                    char ws = (i == NStructReal - 1) ? '\n' : ' ';
+                                    if (i == 0) {
                                         // Multiply mass by 8 since we are only taking 1/8 of the 
                                         // total particles and want to keep the mass in the domain the same.
-                                        File << 8.0* it->m_rdata.arr[i] << ws;
+                                        File << 8.0* it->rdata(i) << ws;
```

Comment:
Looks like a lot of assumptions are made here. 3D only?  The first real being mass?

Reply:
Yes, I agree. This function was moved from Nyx, but it's only used there and really specific to it.

Reply:
I talked to @jmsexton03 and we can remove this function now. I'll do so in a separate PR.